Sources/Widgets/Widgets3D/ResliceCursorWidget/index.js

```diff
@@ -0,0 +1,526 @@
+import macro from 'vtk.js/Sources/macro';
+import vtkAbstractWidgetFactory from 'vtk.js/Sources/Widgets/Core/AbstractWidgetFactory';
+import vtkPlane from 'vtk.js/Sources/Common/DataModel/Plane';
+import vtkPlaneSource from 'vtk.js/Sources/Filters/Sources/PlaneSource';
+import vtkResliceCursorContextRepresentation from 'vtk.js/Sources/Widgets/Representations/ResliceCursorContextRepresentation';
+
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+
+import widgetBehavior from 'vtk.js/Sources/Widgets/Widgets3D/ResliceCursorWidget/behavior';
+import stateGenerator from 'vtk.js/Sources/Widgets/Widgets3D/ResliceCursorWidget/state';
+
+import {
+  getClosestPoint,
+  boundPoint,
+  updateState,
+  getViewPlaneNameFromViewType,
+} from 'vtk.js/Sources/Widgets/Widgets3D/ResliceCursorWidget/helpers';
+import { ViewTypes } from 'vtk.js/Sources/Widgets/Core/WidgetManager/Constants';
+
+import { vec4, mat4 } from 'gl-matrix';
+
+const VTK_INT_MAX = 2147483647;
+const { vtkErrorMacro } = macro;
+
+// ----------------------------------------------------------------------------
+// Factory
+// ----------------------------------------------------------------------------
+
+function vtkResliceCursorWidget(publicAPI, model) {
+  model.classHierarchy.push('vtkResliceCursorWidget');
+
+  model.methodsToLink = ['activeColor', 'useActiveColor', 'opacity'];
+
+  // --------------------------------------------------------------------------
+  // Private methods
+  // --------------------------------------------------------------------------
+
+  function updatePlanePoints(line1, line2, origin, planeSource) {
+    const point1 = getClosestPoint(origin, [
+      line1.getPoint1(),
+      line1.getPoint2(),
+    ]);
+    const point2 = getClosestPoint(origin, [
+      line2.getPoint1(),
+      line2.getPoint2(),
+    ]);
+    planeSource.setPoint1(point1);
```

Comment:
Taking the closest line (i.e.g segment) point as plane source point1 does not seem like the right approach.

Or said differently, I am not sure what that means.

Reply:
For the record, I think it might lead to problems later on, but we can leave it as is for time being